src/app/core/configurable-enum/configurable-enum.interface.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Interface specifying overall object representing an enum with all its options
+ * as stored in the config database
+ */
+export type ConfigurableEnum<
+  T extends ConfigurableEnumValue = ConfigurableEnumValue
```

Comment:
what does this line mean?

Reply:
`ConfigurableEnum` (as the type how an enum collection is stored in the config) is an array of objects implementing `ConfigurableEnumValue` interface by default. But the objects in the config can also have additional properties (like a color, etc.), then e.g. `ConfigurableEnum<InteractionType>` (where `InteractionType` extends the basic `ConfigurableEnumValue`) as a type already also includes type hints for the additional peroperties that InteractionType has.